sympy/calculus/codomain.py

```diff
@@ -151,3 +154,73 @@ def codomain_interval(f, set_val, *sym):
     if isinstance(domain, FiniteSet):
         return FiniteSet(*[limit(func, symbol, i) if i in FiniteSet(-oo, oo)
                             else func.subs({symbol: i}) for i in domain])
+
+
+def not_empty_in(fin_set, intrvl, *syms):
+    """ Finds the domain in which the Set input is not empty
+
+    Parameters
+    ==========
+
+    fin_set: FiniteSet
+            The FiniteSet  not-empty
+    intrvl: Union of Sets
+            The range of the FiniteSet elements
+    syms: Tuple of symbols
+            Symbol for which domain is to be found
+
+    Raises
+    ======
+
+    NotImplementedError
+        The algorithms to find the non-emptiness of the given FiniteSet are
+        not yet implemented.
+    ValueError
+        The input is not valid.
+    RuntimeError
+        It is a bug, please report to the github issue tracker.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, codomain, FiniteSet, Interval, not_empty_in, sqrt, oo
+    >>> x = Symbol('x')
+    >>> not_empty_in(FiniteSet(x/2), Interval(0, 1), x)
+    [0, 2]
+    >>> not_empty_in(FiniteSet(x, x**2), Interval(1, 2), x)
+    [-sqrt(2), -1] U [1, 2]
+    >>> not_empty_in(FiniteSet(x**2/(x + 2)), Interval(1, oo), x)
+    (-2, -1] U [2, oo)
+    """
+
+    # TODO: handle functions with infinitely many solutions (eg, sin, tan)
+    # TODO: handle multivariate functions
+    if not isinstance(fin_set, FiniteSet):
+        raise ValueError('A FiniteSet must be given, not %s: %s' % (type(fin_set), fin_set))
+    if len(syms) == 1:
+        symbol = syms[0]
+    else:
+        raise NotImplementedError('more than one variables %s not handled' % (syms,))
+
+    y = Dummy('y')
+
+    def elm_domain(expr, *sym):
+        domain_union = S.EmptySet
```

Comment:
You might want to write a couple of lines for the doc string. Even though if you make it private it's good to write a line about the functionality.